### PR TITLE
feat: update fiat ramp

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

- Updated fiat ramps to work in the new flow

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a

## Risk

Totally changed the way this works, so its possible there are some edge cases I haven't covered.

## Testing

Test that you can buy/sell assets and it opens the correct provider link.

### Engineering

n/a

### Operations

Test that you can buy/sell assets and it opens the correct provider link.

### CODE REVIEW

File: src/components/Modals/FiatRamps/components/FiatRampActionButtons.tsx [hunk 1]
```diff
@@ -1,48 +1,36 @@
...
-    pl: 10,
-    pr: 10,
-  }
+  const activeIndex = useMemo(() => {
+    if (action === FiatRampAction.Sell) {
+      return 1
+    } else {
+      return 0
+    }
+  }, [action])
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/components/FiatRampActionButtons.tsx:22]:
```suggestion
  const activeIndex = useMemo(() => Number(Boolean(action === FiatRampAction.Sell)), [action])
```

do you like golf beard?

File: src/components/Modals/FiatRamps/components/FiatRampButton.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        >
+          <Flex flexDirection='row' justifyContent='center' alignItems='center'>
+            <AssetIcon src={logo} boxSize='8' />
+            <Box textAlign='left' ml={2}>
+              <Text fontWeight='bold' translation={label} />
+              <Text translation={info ?? ''} />
+            </Box>
+          </Flex>
+          <Flex display={['none', 'flex']} gap={2}>
+            {tags?.map(tag => (
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/components/FiatRampButton.tsx:75]: memo plz

File: src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+const moduleLogger = logger.child({
+  namespace: ['Modals', 'FiatRamps', 'hooks', 'useFiatRampCurrencyList'],
+})
+
+type useFiatRampByAssetIdProps = {
+  assetId: AssetId | undefined
+  action: FiatRampAction
+}
+
+export const useFiatRampByAssetId = ({ assetId, action }: useFiatRampByAssetIdProps) => {
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts:19]: react fc plz

File: src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts [hunk 4]
```diff
@@ -107,16 +107,49 @@ export const useFiatRampCurrencyList = (fiatRampProvider: FiatRamp) => {
...
   useEffect(() => {
     setLoading(true)
-    ;(async () => {
-      const [parsedBuyList, parsedSellList] = await supportedFiatRamps[
-        fiatRampProvider
-      ].getBuyAndSellList()
-      // only the sell list needs balances for sorting
-      setSellList(addSellPropertiesAndSort(parsedSellList))
-      setBuyList(addBuyPropertiesAndSort(parsedBuyList))
+    async function getBySellAssets() {
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts:110]:
```suggestion
    async function getBuySellAssets() {
```

File: src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts [hunk 5]
```diff
@@ -1,4 +1,5 @@
 import { avalancheAssetId } from '@shapeshiftoss/caip'
+import { uniqBy } from 'lodash'
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts:2]:
```suggestion
import uniqBy from 'lodash/uniqBy'
```

at some stage we'll lint rule this for tree shaky reasons

File: src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts [hunk 6]
```diff
@@ -107,16 +107,49 @@ export const useFiatRampCurrencyList = (fiatRampProvider: FiatRamp) => {
...
+    // ;(async () => {
+    //   const [parsedBuyList, parsedSellList] = await supportedFiatRamps[
+    //     fiatRampProvider
+    //   ].getBuyAndSellList()
+    //   // only the sell list needs balances for sorting
+    //   setSellList(addSellPropertiesAndSort(parsedSellList))
+    //   setBuyList(addBuyPropertiesAndSort(parsedBuyList))
+
+    // })()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampCurrencyList.ts:151]: delet

File: src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        if (hasMatch) {
+          acc.push(provider)
+        }
+        return acc
+      }, [])
+      setProviders(parsedProviders)
+      setLoading(false)
+    }
+    getBySellAssets()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts:64]: plz comment what is excluded and why

File: src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+export const useFiatRampByAssetId = ({ assetId, action }: useFiatRampByAssetIdProps) => {
+  const [loading, setLoading] = useState(false)
+  const [providers, setProviders] = useState<SupportedFiatRampConfig[]>([])
+
+  useEffect(() => {
+    if (!assetId) return
+    setLoading(true)
+    async function getBySellAssets() {
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts:26]: i'm not actually sure if this is meant to be `getBy...` or `getBuy...`

File: src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        await Promise.allSettled(
+          Object.values(supportedFiatRamps).map<
+            Promise<
+              [
+                provider: SupportedFiatRampConfig,
+                buyAndSellList: [FiatRampAsset[], FiatRampAsset[]],
+              ]
+            >
+          >(async provider => {
+            return [provider, await provider.getBuyAndSellList()]
```
Comment (0xdef1cafe) [inline src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts:37]: this is going to fire 5 requests every time we switch assets. 

we may want to fire these requests once via an effect, store them in state, and have the hook return a filtered list when the assetId changes?

Comment (reallybeard) [reply src/components/Modals/FiatRamps/hooks/useFiatRampByAssetId.ts:37]: could this be refactored in the fiat ramp abstraction stuff, to read from cache. Otherwise go out and get them?

**Review by 0xdef1cafe — APPROVED** [review]
